src/lib/datasource/abstract.coffee

```diff
@@ -16,8 +20,35 @@ class GridDatasourceAbstract
     return this
 
   ###
+   * Get/Set number of results returned from a query
+   * @param  {number} limit
+   * @returns {this|number}
+  ###
+  limit: ( limit ) ->
+    return @_limit if typeof limit is 'undefined'
```

Comment:
Maybe you should
return @_limit if typeof limit isnt 'number'

So you cover all the invalid cases, not just the undefined.